### DISCUSSION TITLE
Referencing Java classes using SKIP

### DISCUSSION THREAD

**Discussion by pnewell** [skiptools/skip#79] [Q&A] [answered]
Is there a better way to reference the java version of a class other than using SKIP INSERT/REPLACE? Specifically, how can I write the following in Swift? I've tried a couple of different ways and they all end up becoming static references.

```
val mainActivityIntent = Intent(context, MainActivity::class.java)
```

**Comment by marcprux** [comment, marked as answer]
Generally, using `#if SKIP` blocks is cleaner than `// SKIP INSERT:` or `// SKIP REPLACE:` comments, but the code inside needs to be syntactically valid Swift, which will lead to `::class` causing a Swift compile error.

You can work around this with a little bit of casting gymnastics:

```swift
public struct ContentView: View {
    public var body: some View {
        Text("Welcome Skipper!")
        #if SKIP
        let kclass: kotlin.reflect.KClass<ContentView> = ContentView.self
        let intent = android.content.Intent(androidx.compose.ui.platform.LocalContext.current, kclass.java)

        Text("New intent: \(intent)")
        #endif
        Image(systemName: "heart.fill")
            .foregroundColor(.red)
    }
}
```

**Reply by pnewell** [reply]
> ```swift
>         let kclass: kotlin.reflect.KClass<ContentView> = ContentView.self
>         let intent = android.content.Intent(androidx.compose.ui.platform.LocalContext.current, kclass.java)
> ```

Thanks for this, while not as eloquent as I'd like, definitely better than having to do an INSERT/REPLACE. I see the challenge with trying to support this, but might be worth investing in as this is so common when writing Kotlin.

That being said, because of my particular use case, I actually was able to come up with another solution that I'll add here in case it would work for someone in the future. _This requires that you also/already have a variable of whatever class you are trying to reference_.

```swift
let contentView: ContentView = ContentView()
let intent = android.content.Intent(androidx.compose.ui.platform.LocalContext.current, type(of: contentView).java)
```

**Comment by marcprux** [comment]
For the record, the following (nicer) code *won't* work right now:

```swift
let intent = android.content.Intent(androidx.compose.ui.platform.LocalContext.current, ContentView.self.java)
```

This will cause it to be transpiled at:

```swift
val intent = android.content.Intent(androidx.compose.ui.platform.LocalContext.current, (ContentView::class.companionObjectInstance as ContentView.Companion).java)
```

This is because we treat all member access on the class instance as wanting to access static members of the class.
